scripts/stubsabot.py

```diff
@@ -398,6 +398,19 @@ async def main() -> None:
     )
     args = parser.parse_args()
 
+    if args.action_level > ActionLevel.nothing:
+        subprocess.run(["git", "update-index", "--refresh"], capture_output=True)
```

Comment:
what does update-index do?

the way i've done this kind of thing is use `git diff-index HEAD --quiet` which sets exit code to 1 if things are dirty and exit code 0 if things are clean

Reply:
It's the first time I've done this kind of thing, and I was principally working off [this StackOverflow answer](https://stackoverflow.com/a/3879077/13990016) :)

I think the reasoning in the answer sounds pretty good, though. According to the answer:

> if there are files that have been touched, but whose contents are the same as in the index, you'll need to run `git update-index --refresh` before `git diff-index`, otherwise `diff-index` will incorrectly report that the tree is dirty

> the way i've done this kind of thing is use `git diff-index HEAD --quiet` which sets exit code to 1 if things are dirty and exit code 0 if things are clean

I considered that, but playing around with it I decided I liked having a precise error message that would tell me exactly which files had uncommitted changes in them. Don't feel too strongly about it, though :)

Reply:
> the way i've done this kind of thing is use `git diff-index HEAD --quiet` which sets exit code to 1 if things are dirty and exit code 0 if things are clean

I considered that, but playing around with it I decided I liked having a precise error message that would tell me exactly which files had uncommitted changes in them. Don't feel too strongly about it, though :)

Reply:
cool, TIL!